tests/integration/test_create_query_constraints/test.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+import asyncio
+import re
+import random
+import os.path
+from helpers.cluster import ClickHouseCluster
+from helpers.test_tools import assert_eq_with_retry, TSV
+
+cluster = ClickHouseCluster(__file__)
+instance = cluster.add_instance("instance", user_configs=["users.xml"])
+
+
+@pytest.fixture(scope="module", autouse=True)
+def start_cluster():
+    try:
+        cluster.start()
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+
+def test_create_query_constraints():
+
+    expected_error = "Setting max_threads should not be changed"
```

Comment:
Is it only the readonly constraint checked from a parent user?
I mean if the parent user (i.e. the `default` user) has a minmax constraint, and then you create a new user with minmax constraint again then your new constraints must be inside the parent user's minmax range.